FurCoderCXX.1
.TH FurCoderCXX 1 "December, 2006" Linux "User Manual"
.SH NAME:
FurCoderCXX \- Graphical aid for the creation/decrypting of "Fur Codes"
.SH SYNOPSIS:
.B FurCoderCXX
.I [-v, --version]
.SH DESCRIPTION:
The
.B FurCoderCXX
is a graphical application which allows the user to create and/or
decypher the 
.I Code of the Furries.
This is accomplished mainly via menu selection to minimize the number of
user-required entries, and thus possible errors. See the on-line help,
available from either the main menu or the toolbar, for more details on
using this app.
.SH OPTIONS:
.TP
.I \--version (-v):
Display program version, then exit
.SH FILES
.I FurCoderCXX-x.y.z/Documents:
.RS
Support files for the on-line help
.RE
.I FurCoderCXX-x.y.z/etc:
.RS
Application config files
.RE
.I ~/.foxrc/Simo/
.RS
FOX-generated "registry" files for user-specific application settings.
.RE
.SH DIAGNOSTICS:
The
.B FurCoderCXX
includes various message dialogs in case a non-fatal error is discovered. 
This allows the user to correct the error and continue running the app.
Also includes an instrumentation feature to send more diagnostic messages to
standard output to assist in debugging the decode mode. Useful if you get
errors/crashes due to peculiar user input.
.P
There is the possibility of a fatal error that will abort this app. Since
this isn't suppose to happen, these errors don't have a message dialog
associated with them. The nature of the error is written to the standard
error, where it can be viewed if you start the app from a terminal command
line. If you have an apparant crash, restart the app from the terminal to
view these error advisories.

.SH AUTHOR:
The
.B FurCoderCXX
was written by Simo on a 
.B Slackware 10.1
i686 System with the
.B FOX-1.2.3
GUI Toolkit and
.B XEMACS.
.SH REPORT BUGS:
Send bug reports to: simo@furrydolphin.net
.SH COPYRIGHT:
.B FurCoderCXX
is Copyright (C) 2006 by Simo. This program is free software;
you may redistribute it and/or modify it under the terms of the GNU
General Public License as published by the Free Software Foundation;
either version 2 of the license, or (at your option) any later
version. See the on-line help to view a copy of the GPL.

This program is distributed in the hope that it will prove useful,
but WITHOUT ANY WARRANTY; without even an implied warranty as to
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
General Public License for further details.

You may receive a copy of the GNU General Public License from:
Free Software Foundation, Inc., 59 Temple Place, Suite 330,
Boston, MA 02111-1307 USA.

The GPL may also be viewed from the on-line help.

Fur Code Copyright (C) 1998 by Ross Smith

NEWS
-------------------------------------- NEWS --------------------------------------

8 January, 2007:
	Added a "File Print" item to the main menu; added a "Print" button to
	toolbar (with print icon).

6 December, 2006:
	Removed from toolbar the encode/decode check box. Replaced with regular
	toolbar button to toggle modes instead. Added: line edit toolbar buttons
	and main menu commands.

4 August, 2005:
	Simplified the code option command button group. Removed the "uncertain"
	option, moved into option menus.

21 September, 2004:
	Made some changes in the decode mode presentation to improve over-all
	appearance.

19 September, 2004:
	Changed the main window widgets to pack to uniform width. All buttons line
	up evenly now.

30 August, 2004:
	GUI Complete. No changes to report.

29 August, 2004:
	Project start.

-------------------------------------- NEWS --------------------------------------


SourceCode/FurCoderCXX.cpp
//----------------------------- furrycoder.cpp -------------------------------

/***************************************************************************
                            -  Description
                             -------------------
    Begin                : 6 December, 2006
    Copyright            : (C) 2006 by Simo
    E-Mail               : simo (at) furrydolphin (dot) net
    Version              : 1.1.0
 **************************************************************************/

/***************************************************************************
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 *                                                                         *
 *       ----> If software can be free, why can't dolphins? <----          *
 *                                                                         *
 ***************************************************************************

*/
/*
 *        Author: Simo
 */

#include <signal.h>
#include <sys/stat.h>
#include <fcntl.h>
#include "tmainwindow.h"

/*
     Name: main

     Type: Function (int)

     Parameters:
                 argc (int) - Number of command line options
		 argv (Dbl Pointer: char) - Array of command line options
		 Note: FOX parses argv for FOX specific options, passes these
		       along to the FOX application object, leaves user
		       defined options for processing.

     Returns:
              0 --  Indicates success to shell
	      1 --      "     failure  "   "

     Note:
           1) See if user gave the "version" command line option.
	      If yes, print version to stdout and exit

	   2) Create an FXApp object and initialize: use app and vendor
	      names to create a hidden directory in each user's /home dir
	      for application specific settings.

	   3) Create the TMainWindow object and intialize, add signal
	      detect so that <CNTL> C will gracefully terminate the app

	   4) Request X-Server services and enter main event loop.

*/

int main(int argc, char *argv[])
{
//     See if "version" option given on command line

    for (int i= 1; i < argc; i++)
    {
        if (! (strcmp(argv[i], "-v") && strcmp(argv[i], "--version")))
	{
	    printf("FurCoderCXX Version: %s\n", VERSION);
	    return 0;
	}
    }

    FXApp  MyApp("FurCoderCXX", "Simo");
    MyApp.init(argc, argv);
    TMainWindow *Mw=  new TMainWindow(&MyApp);
    MyApp.addSignal(SIGINT, Mw, TMainWindow::ID_EXIT);
    MyApp.create();
    return MyApp.run();
}

//----------------------------- furrycoder.cpp -------------------------------
